src/main/java/roomescape/reservation/presentation_layer/ReservationController.java

```diff
@@ -0,0 +1,66 @@
+package roomescape.reservation.presentation_layer;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import roomescape.reservation.domain_model_layer.ReservationEntity;
+import roomescape.reservation.business_layer.ReservationService;
+
+import java.util.List;
+import java.util.Objects;
+
+@Controller
+@RequiredArgsConstructor
+public class ReservationController {
+    // ReservationDB 객체 주입
+    private final ReservationService reservationService;
+
+    // localhost:8080/reservation 요청 시 예약 페이지가 응답할 수 있도록 구현 (templates/new-reservation.html 연결)
+    @GetMapping("/reservation")
+    public String reservation(){
+        return "new-reservation";
+    }
+
+    // 예약 조회 (localhost:8080/reservations 요청 시 예약자 현황 리스트가 응답할 수 있도록 구현)
+    @GetMapping("/reservations")
+    public ResponseEntity<List<ReservationEntity>> getReservations() {
+        // DB에서 예약 정보를 가져와서 반환
+        List<ReservationEntity> response = reservationService.getReservations();
+        return ResponseEntity.ok()
+                .body(response);
+    }
+
+    // 예약 추가
+    @PostMapping("/reservations")
+    public  ResponseEntity<?> saveReservation(@RequestBody ReservationEntity reservation) {
```

Comment:
```suggestion
    public  ResponseEntity<Void> saveReservation(@RequestBody ReservationEntity reservation) {
```
`?` 대신 `Void` 타입을 사용하시는 건 어떠실까요?! 아니면 따로 `?` 로 반환하도록 작성하신 이유가 있으신 지 궁금합니다 🙂

Reply:
해당 메서드에서 예외적인 case에 대해서는 string타입 에러 메시지를 반환하고 정상적인 경우에는 ReservationEntity를 반환하도록 설계하였습니다-! 그렇기에 메서드 반환 타입을 ? (와일드타입)으로 해주었습니다! 하지만 조금 더 정답에 가까운 코드는 한 메서드에서 2가지 경우를 처리하는 것이 아닌 에러 메시지를 반환하는 경우를 exceptionhandler로 처리해주는 것인 듯합니다!